runtime/compiler/net/ServerStream.hpp

```diff
@@ -0,0 +1,238 @@
+   std::tuple<T...> readCompileRequest()
+      {
+      readBlocking(_cMsg);
+      if (_cMsg.type() == JITServer::MessageType::clientTerminate)
+         {
+         uint64_t clientId = std::get<0>(getRecvData<uint64_t>());
+         throw JITServer::StreamCancel(_cMsg.type(), clientId);
+         }
+      if (_cMsg.version() != 0 && _cMsg.version() != getJITaaSVersion())
+         {
+         throw StreamVersionIncompatible(getJITaaSVersion(), _cMsg.version());
+         }
+      if (_cMsg.type() != JITServer::MessageType::compilationRequest)
+         {
+         throw JITServer::StreamMessageTypeMismatch(JITServer::MessageType::compilationRequest, _cMsg.type());
+         }
+      return getArgs<T...>(_cMsg.mutable_data());
+      }
+   
+   /**
+      @brief Extract the data from the received message and return it
+   */
+   template <typename... T>
+   std::tuple<T...> getRecvData()
+      {
+      return getArgs<T...>(_cMsg.mutable_data());
+      }
+
+   /**
+      @brief Function invoked by server when compilation is completed
+
+      This should be the last message sent by a server as a response to a compilation request.
+      It must include a `statusCode` which can be indicative of an error and could include a 
+      variable number of parameters with compilation artifacts (including the compiled body).
+   */
+   template <typename... T>
+   void finishCompilation(uint32_t statusCode, T... args)
+      {
+      try
+         {
+         write(MessageType::compilationCode, statusCode, args...);
+         }
+      catch (std::exception &e)
+         {
+         if (TR::Options::getVerboseOption(TR_VerboseJITaaS))
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "Could not finish compilation: %s", e.what());
+         }
+      }
+
+   void setClientId(uint64_t clientId)
+      {
+      _clientId = clientId;
+      }
+   uint64_t getClientId() const
+      {
+      return _clientId;
+      }
+
+   /**
+      @brief Function called to deal with incoming connection requests
+
+      This function opens a socket, binds it and then waits for incoming connection
+      requests by using `accept()` in an infinite loop. Once a connection is accepted
+      a ServerStream object is created (receiving the newly opened socket descriptor as
+      a parameter) and passed to the compilation handler. Typically, the compilation
+      handler places the ServerStream object in a queue and returns immediately so that
+      other connection requests can be accepted.
+      Note: because the function does not return, it must be executed on a separate thread.
+
+      @param [in] compiler Object that defines the behavior when a new connection is accepted
+      @param [in] info Pointer to PersistentInfo which contains the port and the timeout value for the connection
+   */
+   static void serveRemoteCompilationRequests(BaseCompileDispatcher *compiler, TR::PersistentInfo *info);
+
+   // Statistics
+   static int _numConnectionsOpened;
```

Comment:
Why are these fields not private?

Reply:
Accessed directly in https://github.com/eclipse/openj9/blob/8f554dac124fcba5518f0225f4f8712692246cf5/runtime/compiler/control/CompilationThread.cpp#L3419

Reply:
Accessor methods could be added without exposing write access.

Reply:
Will add accessor methods